model-optimizer/unit_tests/mo/middle/passes/fusing/fuse_linear_ops_test.py

```diff
@@ -817,6 +817,45 @@ def test_fuse_mul_to_deconv_1(self):
         (flag, resp) = compare_graphs(graph, graph_ref, 'placeholder_1', 'placeholder_1')
         self.assertTrue(flag, resp)
 
+    def test_fuse_mul_data_nodes_names(self):
+        graph = build_graph(nodes_attributes,
+                            [('placeholder_1', 'placeholder_1_data'),
+                             ('placeholder_1_data', 'mul_1'),
+                             ('const_mul_1_w', 'mul_1_w'),
+                             ('mul_1_w', 'mul_1'),
+                             ('mul_1', 'mul_1_data'),
+                             ('mul_1_data', 'conv_1'),
+                             ('const_conv_1_w', 'conv_1_w'),
+                             ('const_conv_1_b', 'conv_1_b'),
+                             ('conv_1_w', 'conv_1'),
+                             ('conv_1_b', 'conv_1'),
+                             ('conv_1', 'conv_1_data'),
+                             ('conv_1_data', 'op_output')
+                             ],
+                            {'placeholder_1_data': {'shape': np.array([1, 227, 227, 3])},
+                             'mul_1_data': {'shape': np.array([1, 227, 227, 3])},
+                             'const_mul_1_w': {'shape': np.array([3]), 'value': np.array([1, 2, 3])},
+                             'mul_1_w': {'shape': np.array([3]), 'value': np.array([1, 2, 3])},
+                             'const_conv_1_w': {'shape': np.array([11, 11, 3, 96]),
+                                                'value': np.ones((11, 11, 3, 96))},
+                             'conv_1_w': {'shape': np.array([11, 11, 3, 96]), 'value': np.ones((11, 11, 3, 96)),
+                                          'output_channel_dim': 3, 'input_channel_dim': 2,
+                                          'dims_number': 4},
+                             'const_conv_1_b': {'shape': np.array([96]), 'value': np.zeros(96)},
+                             'conv_1_b': {'shape': np.array([96]), 'value': np.zeros(96)},
+                             'conv_1_data': {}
+                             })
+
+        _fuse_mul(graph, Node(graph, 'mul_1'), [Node(graph, 'conv_1')], backward=False)
```

Comment:
Update the test and add the `can_be_fused` attribute to the edge. Add an assert checking that the attribute still present. Add a comment to the test describing why we care about the edge attribute.

Reply:
The attributes that were lost and caused the issue were actually 'output_channel_dim', 'input_channel_dim' and 'dims_number' on the data node. I will include them to unit test and 'can_be_fused' too just in case.